pkg/graphviz/traceflow_test.go

```diff
@@ -181,6 +187,216 @@ func TestGenGraph(t *testing.T) {
+					UDP: &crdv1alpha1.UDPHeader{
+						SrcPort: 68,
+						DstPort: 80,
+					},
+				},
+				Length: 84,
+				SrcIP:  "10.10.1.4",
+			},
+			Phase: crdv1alpha1.Succeeded,
+			Results: []crdv1alpha1.NodeResult{
+				{
+					Node: "k8s-node-worker-1",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:    crdv1alpha1.ActionForwarded,
+							Component: crdv1alpha1.ComponentSpoofGuard,
+						},
+						{
+							Action:        crdv1alpha1.ActionForwarded,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+							TunnelDstIP:   "192.168.77.100",
+						},
+					},
+				},
+				{
+					Node: "k8s-node-control-plane",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:        crdv1alpha1.ActionReceived,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Classification",
+						},
+						{
+							Action:        crdv1alpha1.ActionDropped,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	liveTfwithNoEndpointNodename := crdv1alpha1.Traceflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-live-traceflow-no-endpoint-nodename",
+		},
+		Spec: crdv1alpha1.TraceflowSpec{
+			LiveTraffic: true,
+			Packet: crdv1alpha1.Packet{
+				IPHeader: crdv1alpha1.IPHeader{
+					Protocol: 1,
+				},
+				TransportHeader: crdv1alpha1.TransportHeader{
+					ICMP: &crdv1alpha1.ICMPEchoRequestHeader{
+						ID:       123,
+						Sequence: 1,
+					},
+				},
+			},
+		},
+		Status: crdv1alpha1.TraceflowStatus{
+			CapturedPacket: &crdv1alpha1.Packet{
+				DstIP: "10.10.0.2",
+				IPHeader: crdv1alpha1.IPHeader{
+					Flags:    2,
+					Protocol: 1,
+					TTL:      63,
+				},
+				TransportHeader: crdv1alpha1.TransportHeader{
+					TCP: &crdv1alpha1.TCPHeader{
+						SrcPort: 123,
+						DstPort: 1,
+						Flags:   2,
+					},
+				},
+				Length: 84,
+				SrcIP:  "10.10.1.4",
+			},
+			Phase: crdv1alpha1.Succeeded,
+			Results: []crdv1alpha1.NodeResult{
+				{
+					Node: "k8s-node-worker-1",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:    crdv1alpha1.ActionForwarded,
+							Component: crdv1alpha1.ComponentSpoofGuard,
+						},
+						{
+							Action:        crdv1alpha1.ActionForwarded,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+							TunnelDstIP:   "192.168.77.100",
+						},
+					},
+				},
+				{
+					Node: "k8s-node-control-plane",
+					Observations: []crdv1alpha1.Observation{
+						{
+							Action:        crdv1alpha1.ActionReceived,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Classification",
+						},
+						{
+							Action:        crdv1alpha1.ActionDelivered,
+							Component:     crdv1alpha1.ComponentForwarding,
+							ComponentInfo: "Output",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	NoliveTf := crdv1alpha1.Traceflow{
```

Comment:
Don't capitalize the variable if you don't need to export it. 
```suggestion
	noLiveTf := crdv1alpha1.Traceflow{
```